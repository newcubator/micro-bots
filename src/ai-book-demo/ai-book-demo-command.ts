import { APIGatewayEvent } from "aws-lambda";
import { ActionType } from "../slack/types/slack-types";

export const commandHandler = async (event: APIGatewayEvent) => {
  return {
    statusCode: 200,
    body: JSON.stringify({
      blocks: [
        {
          dispatch_action: true,
          type: "input",
          block_id: "BOOK_SUPPORT_TEXT",
          element: {
            type: "plain_text_input",
            action_id: ActionType.BOOK_SUPPORT,
          },
          label: {
            type: "plain_text",
            text: "Ich beantworte dir gerne Fragen zum Book! Was m√∂chtest du wissen?",
            emoji: true,
          },
        },
      ],
    }),
  };
};
