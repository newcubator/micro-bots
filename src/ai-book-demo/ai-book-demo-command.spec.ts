import { commandHandler } from "./ai-book-demo-command";

test("command handler", async () => {
  const result = await commandHandler({} as any);
  expect(result.statusCode).toBe(200);
  expect(result.body).toMatchSnapshot();
});
