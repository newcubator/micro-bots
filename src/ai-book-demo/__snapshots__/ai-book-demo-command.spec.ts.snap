// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command handler 1`] = `"{"blocks":[{"dispatch_action":true,"type":"input","block_id":"BOOK_SUPPORT_TEXT","element":{"type":"plain_text_input","action_id":"BOOK_SUPPORT"},"label":{"type":"plain_text","text":"Ich beantworte dir gerne Fragen zum Book! Was m√∂chtest du wissen?","emoji":true}}]}"`;
